sys/log/full/src/log.c

```diff
@@ -492,6 +494,18 @@ log_append_prepare(struct log *log, uint8_t module, uint8_t level,
 #else
     assert(etype == LOG_ETYPE_STRING);
 #endif
+#if MYNEWT_VAL(LOG_VERSION) > 3
+
+#if MYNEWT_VAL(LOG_FLAGS_LOG_IMG_HASH)
+    uint8_t hash[IMGMGR_HASH_LEN];
+
+    ue->ue_flags = LOG_FLAGS_LOG_IMG_HASH_e;
```

Comment:
ue->ue_flags should be set by left shifting `1` by `LOG_FLAGS_LOG_IMG_HASH_e` and logically anding it with the flags field.
Eg: `ue->ue_flags &= (1 << LOG_FLAGS_LOG_IMG_HASH_e)`
What we generally do is define the flags such that they are already left shifted by 1 and are not enums.

Eg:
Definition:
```
typedef enum {
    LOG_FLAGS_LOG_IMG_HASH_e = 1,
    LOG_FLAGS_MAX_e,
}log_flags_e;
```
would be
```
#define LOG_FLAGS_LOG_IMG_HASH           (1 << 0)
#define LOG_FLAGS_LOG_FUTURE_YYY1      (1 << 1)  **/* Just an example, do not add this */**
#define LOG_FLAGS_LOG_FUTURE_YYY2     (1 << 2)  **/* Just an example, do not add this */**
```
And so on. That way you do not have to do the shifting in the code and the above code snippet would be:
`ue->ue_flags &= LOG_FLAGS_LOG_IMG_HASH`

Reply:
Moved to bit shift #define as opposed to enums.